cdap-ui/app/cdap/components/SecureKeys/SecureKeyList/index.tsx

```diff
@@ -93,25 +99,34 @@ interface ISecureKeyListProps extends WithStyles<typeof styles> {
   state: any;
   alertSuccess: () => void;
   alertFailure: () => void;
-  openDeleteDialog: (index: number) => void;
   openEditDialog: (index: number) => void;
+  openDeleteDialog: (index: number) => void;
+  onSecureKeyDetailsClick: (index: number) => void;
 }
 
 const SecureKeyListView: React.FC<ISecureKeyListProps> = ({
   classes,
   state,
   alertSuccess,
   alertFailure,
-  openDeleteDialog,
   openEditDialog,
+  openDeleteDialog,
+  onSecureKeyDetailsClick,
 }) => {
   const { secureKeys } = state;
 
+  // used for filtering down secure keys
+  const [searchText, setSearchText] = React.useState('');
+
   const [createDialogOpen, setCreateDialogOpen] = React.useState(false);
 
+  const filteredSecureKeys = secureKeys.filter(
+    (key) => key.get('name').includes(searchText) || key.get('description').includes(searchText)
```

Comment:
you might want to normalize the key and searchText first (ie. lower case them) so that the search is case insensitive

Reply:
done